packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -235,9 +250,12 @@ export class BucketDeployment extends CoreConstruct {
         SystemMetadata: mapSystemMetadata(props),
         DistributionId: props.distribution?.distributionId,
         DistributionPaths: props.distributionPaths,
+        // Passing through the ARN sequences dependencees on the deployment
+        PassThroughBucketArn: props.destinationBucket.bucketArn,
       },
     });
 
+    this.deployedBucket = s3.Bucket.fromBucketArn(this, 'DestinationBucket', Token.asString(deployment.getAtt('DestinationBucketArn')));
```

Comment:
Pretty sure we're missing a unit test for actually using this property in another construct.